### PR TITLE
raise hotbar limit to 32 slots, add associated keybinding options

### LINKED ISSUE #7915: [Request] Raise HUD hotbar limit to 32 slots
https://github.com/minetest/minetest/blob/b298b0339c79db7f5b3873e73ff9ea0130f05a8a/src/hud.h#L53

This sets an arbitrary limit, which aside from my screen being able to handle more, is an unusual value given that the default inventory is an 8x4 grid (so why 23, instead of 24 or 32?).

Changing this value alone is enough to allow more hotbar slots (I checked up to 28, the limit of my screen at my usual GUI scaling).

There are other places in the code that reference a hardcoded value of 23, without referring to this specific `#define`, for example these key assignments:
https://github.com/minetest/minetest/blob/b298b0339c79db7f5b3873e73ff9ea0130f05a8a/src/client/keys.h#L76-L99
https://github.com/minetest/minetest/blob/b298b0339c79db7f5b3873e73ff9ea0130f05a8a/src/client/inputhandler.cpp#L78-L81
https://github.com/minetest/minetest/blob/5f1cd555cd9d1c64426e173b30b5b792d211c835/src/client/game.cpp#L2004-L2012
(thanks goes to @kkabrams for finding these)

Anyway, I'd like to request that this be raised to 32 in both 0.4.x and 5.0.0.

@kahrl you set this value several years ago, with commit 96fe1de83.

### PR DESCRIPTION
**Pull request by VanessaE**
Closes #7915

### CODE REVIEW

**Comment by paramat** [comment]
I support raising the limit to 32, even if not enough keys exist to bind them all.
Not for 0.4.x though, it's not a bugfix, and i know you're trying to resist moving to 5.0.0. The more attractive 5.0.0 is the better.

**Comment by ClobberXD** [comment]
Please also update the documentation here:

https://github.com/minetest/minetest/blob/dcf58a3ad0c1092b3537039ee55892bb7dc83dee/doc/lua_api.txt#L5188-L5189

**Comment by ClobberXD** [comment]
`lua_api.txt` also needs to be updated here:

https://github.com/minetest/minetest/blob/dcf58a3ad0c1092b3537039ee55892bb7dc83dee/doc/lua_api.txt#L5188-L5189

File: src/defaultsettings.cpp [hunk 1]
```diff
@@ -125,6 +125,15 @@ void set_default_settings(Settings *settings)
...
 	settings->setDefault("keymap_slot23", "");
+	settings->setDefault("keymap_slot24", "");
+	settings->setDefault("keymap_slot25", "");
+	settings->setDefault("keymap_slot26", "");
+	settings->setDefault("keymap_slot27", "");
+	settings->setDefault("keymap_slot28", "");
+	settings->setDefault("keymap_slot29", "");
+	settings->setDefault("keymap_slot30", "");
+	settings->setDefault("keymap_slot31", "");
+	settings->setDefault("keymap_slot32", "");
```
Comment (SmallJoker) [inline src/defaultsettings.cpp]: Please adjust `settingtypes.txt` accordingly.

**Comment by VanessaE** [comment]
Fixed docs and settingtypes.txt.

File: src/client/game.cpp [hunk 2]
```diff
@@ -2001,7 +2002,7 @@ void Game::processItemSelection(u16 *new_playeritem)
 
 	/* Item selection using hotbar slot keys
 	 */
-	for (u16 i = 0; i < 23; i++) {
+	for (u16 i = 0; i < HUD_HOTBAR_ITEMCOUNT_MAX; i++) {
```
Comment (SmallJoker) [inline src/client/game.cpp]:
```suggestion
	for (u16 i = 0; i <= max_item; i++) {
```
So you can get rid of the unnecessary range check two lines below.

Comment (VanessaE) [reply src/client/game.cpp]: Sounds okay I guess, but can't `max_item` sometimes have a different value than `HUD_HOTBAR_ITEMCOUNT_MAX` ?

C++ definitely isn't my forte. :smiley:

**Pull request merged**